### PR TITLE
refactor(server): e2e

### PR DESCRIPTION
**Pull request by jrasm91**
* Turn on microservices in e2e tests
* Add support for awaiting webhook events for async tasks
* Migrate trash e2e
* Refactor cli.upload spec to not require a full database reset (runs much faster)
* Migrate a few asset endpoints to e2e

### CODE REVIEW

File: e2e/src/utils.ts [hunk 1]
```diff
@@ -170,10 +177,42 @@ export interface AdminSetupOptions {
...
+    return new Promise<T>((resolve, reject) => {
+      const timeout = setTimeout(() => reject(new Error('Timeout')), 4000);
+      ws.once(event, (data: T) => {
+        clearTimeout(timeout);
+        resolve(data);
+      });
+    });
+  },
+};
+
```
Comment (jrasm91) [inline e2e/src/utils.ts:210]: This is the new websocket stuff

File: e2e/src/api/specs/trash.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    it('should empty the trash', async () => {
+      const { id: assetId } = await apiUtils.createAsset(admin.accessToken);
+      await apiUtils.deleteAssets(admin.accessToken, [assetId]);
+
+      const before = await getAllAssets(
+        {},
+        { headers: asBearerAuth(admin.accessToken) },
+      );
+
+      expect(before.length).toBeGreaterThanOrEqual(1);
```
Comment (bo0tzz) [inline e2e/src/api/specs/trash.e2e-spec.ts:40]: Should this be an exact equals?

**Review by bo0tzz — APPROVED** [review]
Love it

File: e2e/src/api/specs/trash.e2e-spec.ts (hunk 2 above)
Comment (jrasm91) [reply e2e/src/api/specs/trash.e2e-spec.ts:40]: The tests don't reset the database between test cases so there are some assets already there

File: e2e/src/utils.ts [hunk 3]
```diff
@@ -59,7 +66,7 @@ let client: pg.Client | null = null;
 export const fileUtils = {
   reset: async () => {
     await execPromise(
-      `docker exec -i "${serverContainerName}" rm -R "${uploadMediaDir}"`
+      `docker exec -i "${serverContainerName}" /bin/bash -c "rm -rf ${dirs} && mkdir ${dirs}"`,
```
Comment (danieldietzler) [inline e2e/src/utils.ts:69]: When bun shell commands? :D

**Review by danieldietzler — COMMENTED** [review]
Nice!

**Pull request merged**